toolchain/driver/compile_subcommand.cpp

```diff
@@ -465,7 +465,21 @@ class CompilationUnit {
 
     bool print = options_.dump_sem_ir && IncludeInDumps();
     if (vlog_stream_ || print) {
-      SemIR::Formatter formatter(*tokens_, *parse_tree_, *sem_ir_);
+      // Omit entities imported from files that we are not dumping.
+      auto should_format_entity = [&](SemIR::InstId entity_inst_id) -> bool {
+        auto loc_id = sem_ir_->insts().GetLocId(entity_inst_id);
+        if (!loc_id.is_import_ir_inst_id()) {
+          return true;
+        }
+        auto import_ir_id =
+            sem_ir_->import_ir_insts().Get(loc_id.import_ir_inst_id()).ir_id;
+        const auto* import_file =
+            sem_ir_->import_irs().Get(import_ir_id).sem_ir;
+        return !import_file || IncludeInDumps(import_file->filename());
```

Comment:
What's the case where we store an import with a null `sem_ir`? I didn't think that was allowed. Maybe one of:

```suggestion
        const auto* import_file =
            sem_ir_->import_irs().Get(import_ir_id).sem_ir;
        return IncludeInDumps(import_file->filename());
```

```suggestion
        const auto* import_file =
            sem_ir_->import_irs().Get(import_ir_id).sem_ir;
        CARBON_CHECK(import_file);
        return IncludeInDumps(import_file->filename());
```

Reply:
Yeah, I think this can't happen -- definitely not for an IR that we've imported something from.